x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.ts

```diff
@@ -315,6 +328,56 @@ export const retrieveListOfPoliciesIfNeeded = async (
   }
 };
 
+const fetchEditTrustedAppIfNeeded = async (
+  { getState, dispatch }: ImmutableMiddlewareAPI<TrustedAppsListPageState, AppAction>,
+  trustedAppsService: TrustedAppsService
+) => {
+  const currentState = getState();
+  const isPageActive = trustedAppsListPageActive(currentState);
+  const isEditFlow = isEdit(currentState);
+  const isAlreadyFetching = isFetchingEditTrustedAppItem(currentState);
+  const editTrustedAppId = editItemId(currentState);
+
+  if (isPageActive && isEditFlow && editTrustedAppId && !isAlreadyFetching) {
+    let trustedAppForEdit = editingTrustedApp(currentState);
+
+    // If Trusted App is already loaded, then do nothing
+    if (trustedAppForEdit && trustedAppForEdit.id === editTrustedAppId) {
+      return;
+    }
+
+    // See if we can get the Trusted App record from the current list of Trusted Apps being displayed
+    trustedAppForEdit = getListItems(currentState).find((ta) => ta.id === editTrustedAppId);
+    if (trustedAppForEdit) {
+      dispatch({
+        type: 'trustedAppCreationEditItemStateChanged',
+        payload: {
+          type: 'LoadedResourceState',
+          data: trustedAppForEdit,
+        },
+      });
+
+      dispatch({
+        type: 'trustedAppCreationDialogFormStateUpdated',
+        payload: {
+          entry: toNewTrustedApp(trustedAppForEdit),
+          isValid: true,
+        },
+      });
+      return;
+    }
+
+    // Retrieve Trusted App record via API. This would be the case when linking from another place or
+    // using an UUID for a Trusted App that is not currently displayed on the list view.
+
+    // eslint-disable-next-line no-console
+    console.log('todo: api call');
+
+    // FIXME: Implement GET API - https://github.com/elastic/security-team/issues/682#issuecomment-768581449
```

Comment:
same here

Reply:
take out the URL, leave the comment

![cannoli.gif](https://i.giphy.com/media/3o7bu4ZLGQTSBRJIBi/giphy.webp)

Reply:
OHhhhh. Right 🤦